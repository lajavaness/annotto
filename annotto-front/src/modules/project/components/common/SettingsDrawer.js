import { Form, Switch } from 'antd'
import { useTranslation } from 'react-i18next'
import PropTypes from 'prop-types'
import React from 'react'

import * as Styled from './__styles__/SettingsDrawer.styles'

const SettingsDrawer = ({
	visible,
	highlights,
	isHighlightAllowed,
	showPredictions,
	prefillPredictions,
	onClose,
	onValuesChange,
}) => {
	const { t } = useTranslation('project')

	return (
		<Styled.Drawer
			placement="left"
			width={320}
			closable={false}
			onClose={onClose}
			visible={visible}
			getContainer={false}
		>
			<Styled.Row>
				<Styled.Header>{t('project:projectSettings.title')}</Styled.Header>
				<Styled.LeftCircleFilled onClick={onClose} />
			</Styled.Row>
			<Form
				layout="vertical"
				initialValues={{ highlights, showPredictions, prefillPredictions }}
				onValuesChange={onValuesChange}
			>
				<Form.Item name="showPredictions" label={t('project:projectSettings.showPredictions')} valuePropName="checked">
					<Switch />
				</Form.Item>
				<Form.Item
					name="prefillPredictions"
					label={t('project:projectSettings.prefillPredictions')}
					valuePropName="checked"
				>
					<Switch />
				</Form.Item>
				{isHighlightAllowed && (
					<Form.Item name="highlights" label={t('project:projectSettings.highlights')}>
						<Styled.Select allowClear mode="tags" dropdownStyle={{ display: 'none' }} value={highlights} />
					</Form.Item>
				)}
			</Form>
		</Styled.Drawer>
	)
}

export default SettingsDrawer

SettingsDrawer.propTypes = {
	/** Defines if the drawer is visible. */
	visible: PropTypes.bool,
	/** Defines the highlighted of the project. */
	highlights: PropTypes.arrayOf(PropTypes.string),
	/** Defines if the highlight is allowed. */
	isHighlightAllowed: PropTypes.bool,
	/** Defines the value of showPredictions in project. */
	showPredictions: PropTypes.bool,
	/** Defines the value of prefillPredictions in project. */
	prefillPredictions: PropTypes.bool,
	/** Defines the handler called when the close button is clicked. */
	onClose: PropTypes.func,
	/** Defines the handler called when the form is changed. */
	onValuesChange: PropTypes.func,
}

SettingsDrawer.defaultProps = {
	visible: false,
	highlights: null,
	isHighlightAllowed: false,
	showPredictions: false,
	prefillPredictions: false,
	onClose: null,
	onValuesChange: null,
}
