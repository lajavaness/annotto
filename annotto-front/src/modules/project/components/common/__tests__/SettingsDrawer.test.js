import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import React from 'react'

import theme from '__theme__'

import SettingsDrawer from 'modules/project/components/common/SettingsDrawer'

const defaultProps = {}
const getInstance = (props = {}) => (
	<ThemeProvider theme={theme}>
		<SettingsDrawer {...defaultProps} {...props} />
	</ThemeProvider>
)

describe('SettingsDrawer', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(getInstance())
		expect(asFragment()).toMatchSnapshot()
	})
})
