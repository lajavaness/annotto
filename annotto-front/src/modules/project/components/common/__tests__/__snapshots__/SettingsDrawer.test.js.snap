// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsDrawer matches snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div
      class="ant-drawer ant-drawer-left sc-gsBqHK cQSSJx"
      tabindex="-1"
    >
      <div
        class="ant-drawer-mask"
      />
      <div
        class="ant-drawer-content-wrapper"
        style="transform: translateX(-100%); width: 320px;"
      >
        <div
          class="ant-drawer-content"
        >
          <div
            class="ant-drawer-wrapper-body"
          >
            <div
              class="ant-drawer-body"
            >
              <div
                class="ant-row sc-aKXVD bJlwGZ"
              >
                <div
                  class="sc-kmATbt hPoARU"
                >
                  project:projectSettings.title
                </div>
                <span
                  aria-label="left-circle"
                  class="anticon anticon-left-circle sc-cvJIAo dDDzUF"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="left-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm104 316.9c0 10.2-4.9 19.9-13.2 25.9L457.4 512l145.4 105.2c8.3 6 13.2 15.6 13.2 25.9V690c0 6.5-7.4 10.3-12.7 6.5l-246-178a7.95 7.95 0 010-12.9l246-178a8 8 0 0112.7 6.5v46.8z"
                    />
                  </svg>
                </span>
              </div>
              <form
                class="ant-form ant-form-vertical"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="showPredictions"
                      title="project:projectSettings.showPredictions"
                    >
                      project:projectSettings.showPredictions
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <button
                          aria-checked="false"
                          class="ant-switch"
                          id="showPredictions"
                          role="switch"
                          type="button"
                        >
                          <div
                            class="ant-switch-handle"
                          />
                          <span
                            class="ant-switch-inner"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="prefillPredictions"
                      title="project:projectSettings.prefillPredictions"
                    >
                      project:projectSettings.prefillPredictions
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <button
                          aria-checked="false"
                          class="ant-switch"
                          id="prefillPredictions"
                          role="switch"
                          type="button"
                        >
                          <div
                            class="ant-switch-handle"
                          />
                          <span
                            class="ant-switch-inner"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
